src/utils/destination-configuration.ts

```diff
@@ -0,0 +1,195 @@
+import axios, { AxiosInstance } from 'axios';
+import { setupAuthorizationHeaderOnRequestInterceptor } from './axios-utils';
+import { BasicAuthentication, OAuthAuthentication } from '../authentication';
+import { KeyStore, KeystoreFormats } from './key-store';
+import jks = require('jks-js');
+
+export interface Credentials {
+    /**
+     * Username
+     */
+    username: string;
+    /**
+     * Password
+     */
+    password: string;
+    /**
+     * Destination name
+     */
+    destinationName: string;
+    /**
+     * Destination url to call on retrieving destination
+     */
+    destinationUrl: string;
+    /**
+     * OAuth token url to call on retrieving token
+     */
+    oAuthTokenUrl: string;
+}
+
+/**
+ * Destination Service Configuration class. Retrieves the destination from destination service and build authentication.
```

Comment:
`Destination Service Configuration class. Retrieves the destination from Destination service and build authentication contenxt`